samples/experimental/skills-prototypes/dialog-to-dialog/dialog-root-bot/adapter_with_error_handler.py

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+import sys
+import traceback
+from datetime import datetime
+
+from botbuilder.core import (
+    BotFrameworkAdapter,
+    BotFrameworkAdapterSettings,
+    ConversationState,
+    MessageFactory,
+    TurnContext,
+)
+from botbuilder.schema import ActivityTypes, Activity, InputHints
+
+
+class AdapterWithErrorHandler(BotFrameworkAdapter):
+    def __init__(
+        self,
+        settings: BotFrameworkAdapterSettings,
+        conversation_state: ConversationState = None,
+    ):
+        super().__init__(settings)
+        self._conversation_state = conversation_state
+
+        # Catch-all for errors.
+        async def on_error(context: TurnContext, error: Exception):
```

Comment:
Take a look at the latest c# implementation, we broke this handler into separate methods to make it easier to read and we also added try/catch/swallow in several blocks to ensure the bot continues running even if it fails to send traces.
https://github.com/microsoft/botbuilder-dotnet/blob/master/FunctionalTests/Skills/SimpleBotToBot/SimpleRootBot/AdapterWithErrorHandler.cs

Reply:
This was fixed in #853